### PR TITLE
Fix QAT resume with BN models, checkpoint name

### PR DESCRIPTION
**Pull request by oguzhanbsolak**
I noticed these bugs recently and this pr is independent of the bug fixes and updates I mentioned in last week's meeting.

image

When resuming a checkpoint, a new optimizer is created from the model. As batchnorm fusing reduces models' parameter sizes, when resuming from a qat_checkpoint new optimizer have less parameters than the optimizer state_dict at the checkpoint.
Therefore, I added a update_optimizer function to call at initiate_qat state to strip off the batchnorm parameters from the optimizer. 

Also, after resuming a qat_checkpoint, checkpoint names were incorrect. This pr includes a simple fix for that as well.

Limitations:
- This solution is tested for a single params_group case. As our current train.py structure always produces a single params_group, it should be okay.  However, if we ever add functionality to have different learning rates for different layers, we should check this function as well.

### CODE REVIEW

**Review by ermanok — APPROVED** [review]
Looks good.

**Review by rotx-maxim — APPROVED** [review]
